libsolutil/CDCL.cpp

```diff
@@ -45,6 +46,8 @@ CDCL::CDCL(
 
 optional<CDCL::Model> CDCL::solve()
 {
+	if (!ok) return nullopt;
+
```

Comment:
Doesn't this need some preprocessing here to enforce the 1-/2-long clause invariant?

Reply:
I enforce it at `addClause` and after analysis. They are the only places, AFAIK, that can add clauses to the DB. Maybe I missed a place? :S

Reply:
As we discussed out-of-band, thankfully this needs no check here, due to both addClause enforcing the invariant, and the learnt clause adding enforcing the invariant, too. I am resolving. Please re-open in case there are some concerns.